src/common/hashtable/src/hashjoin_hashtable.rs

```diff
@@ -109,7 +109,7 @@ where
 
     fn contains(&self, key_ref: &Self::Key) -> bool {
         let index = key_ref.hash() as usize & self.hash_mask;
-        let mut raw_entry_ptr = self.pointers[index];
```

Comment:
Do we really need unsafe?

Reply:
> Do we really need unsafe?

The index here must be safe, so we can use `get_unchecked` to get a very small performance improvement. 😅

Reply:
I think we should use unsafe as little as possible, usually adding unsafe gives zero perf gain, compiler can do many things. But unsafe can increase vulnerability of our code.

Even if there really might be a performance gain, we should have a separate PR with only unsafe modifications to ensure that the principle of singularity proves whether or not it's a performance gain from unsafe.

Reply:
Yes, but here we use hash & mask to get the index, it must be safe, I will add some #Safety comments, because the probe_hash_table will be called many times, it is worth doing.